core/commands/repo.go

```diff
@@ -95,3 +97,82 @@ order to reclaim hard disk space.
 		},
 	},
 }
+
+type RepoLsOutput struct {
+	Hash string
+	Size uint64
+	Type string
+}
+
+var repoLsCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Totally need a documentation",
+		ShortDescription: `
+		I might agree on that
+		`,
+	},
+
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		outChan := make(chan interface{})
+		res.SetOutput((<-chan interface{})(outChan))
+
+		go func() {
+			defer close(outChan)
+			for _, k := range n.Pinning.RecursiveKeys() {
+				node, err := n.DAG.Get(req.Context(), k)
+				if err != nil {
+					res.SetError(err, cmds.ErrNormal)
+					return
+				}
+
+				unixFSNode, err := unixfs.FromBytes(node.Data);
```

Comment:
note that things other than unixfs nodes may be pinned, probably want to include some way to distinguish between unixfs and 'other'